src/plugin.ts

```diff
@@ -0,0 +1,103 @@
+import { FastifyLoggerInstance, FastifyPluginAsync } from 'fastify';
+import { Actor, Item, Member, PermissionLevel, UnknownExtra } from 'graasp';
+import { ItemTagTaskManager, ItemTagService } from 'graasp-item-tags';
+import { isGraaspError } from './utils';
+
+export interface GraaspHiddenOptions {
+  hiddenTagId: string;
+}
+
+const plugin: FastifyPluginAsync<GraaspHiddenOptions> = async (fastify, options) => {
+  const {
+    items: { dbService: iS, taskManager: itemTaskManager },
+    itemMemberships: { dbService: iMS, taskManager: membershipTaskManager },
+    taskRunner: runner,
+  } = fastify;
+
+  const { hiddenTagId } = options;
+
+  const iTS = new ItemTagService();
+  const taskManager = new ItemTagTaskManager(iS, iMS, iTS, itemTaskManager);
+
+  const isItemHidden = async (item: Item, actor: Actor, log: FastifyLoggerInstance) => {
+    const t1 = taskManager.createGetOfItemTask(actor as Member, item);
+    const t2 = membershipTaskManager.createGetMemberItemMembershipTask(actor, {
+      item,
+      validatePermission: 'admin' as PermissionLevel,
```

Comment:
I doesn't work if used with PermissionLevel.Admin?

Reply:
No, if PermissionLevel.Admin, when running the plugin cannot find the graasp module